src/jarabe/desktop/viewtoolbar.py

```diff
@@ -147,6 +147,8 @@ def _entry_activated_cb(self, entry):
         new_query = entry.props.text
         if self._query != new_query:
             self._query = new_query
+            if isinstance(self._query, bytes):
```

Comment:
@pro-panda   why did you add this line in the code,because it not  not related to this issue of porting to a different version @quozl Your  suggestions will also be helpful

Reply:
@manddy, thanks for asking.  The change seems common to me.  @pro-panda didn't explain the change in the commit message.  I'm not sure why the change is needed though.  I've no suggestions.

Reply:
Thanks for asking @manddy.
Most of such changes were introduced since https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/383 supported both Python 2 and 3, which implied that you could expect `bytes` or `unicode`, depending on the version an interacting sugar activity is written in.

From what I recall, this instance was also implemented for a similar reason.

Reply:
Oh, of course, thanks.